### PR TITLE
feat(mutators): Add string.IsNullOrEmpty and string.IsNullOrWhiteSpace mutations

### LINKED ISSUE #1581: Mutator idea: string.IsNullOrEmpty
**Is your feature request related to a problem? Please describe.**
When using string.IsNullOrEmpty most of the times only null is tested. 

**Describe the solution you'd like**
If we mutate `string.IsNullOrEmpty(x)` into `(x != null)` and `(x != string.Empty)` it's clear both are tested.

### PR DESCRIPTION
**Pull request by yakivyusin**
Closes #1581

### CODE REVIEW

File: docs/mutations.md [hunk 1]
```diff
@@ -161,6 +161,11 @@ Do you have a suggestion for a (new) mutator? Feel free to create an [issue](htt
 | `$"foo {bar}"` | `$""` |
 | `@"foo"` | `@""` |
 | `string.Empty` | `"Stryker was here!"` |
+| `string.IsNullOrEmpty(x)` | `x != null` |
+| `string.IsNullOrEmpty(x)` | `x != ""` |
+| `string.IsNullOrWhiteSpace(x)` | `x != null` |
+| `string.IsNullOrWhiteSpace(x)` | `x != ""` |
+| `string.IsNullOrWhiteSpace(x)` | `x.Trim().Length != 0` |
```
Comment (psfinaki) [inline docs/mutations.md:168]: TBH That looks a bit off here, what's the justification for having it?

File: src/Stryker.Core/Stryker.Core/Mutants/CsharpMutantOrchestrator.cs [hunk 2]
```diff
@@ -51,7 +51,8 @@ public CsharpMutantOrchestrator(IEnumerable<IMutator> mutators = null, StrykerOp
                 new NullCoalescingExpressionMutator(),
                 new MathMutator(),
                 new SwitchExpressionMutator(),
-                new IsPatternExpressionMutator()
+                new IsPatternExpressionMutator(),
+                new StringIsNullOrMutator()
```
Comment (psfinaki) [inline src/Stryker.Core/Stryker.Core/Mutants/CsharpMutantOrchestrator.cs:55]: StringIsNullOrEmptyMutator? Also, I believe this should be just part of the StringEmptyMutator somehow.

File: docs/mutations.md (hunk 1 above)
Comment (yakivyusin) [reply docs/mutations.md:168]: Do you mean the `x.Trim().Length` mutation? This is check that the whitespace-only string is covered by unit tests. `Trim()` will return the zero-length string for the whitespace-only input.

This mutation also can be rewritten as `x.Trim() != ""` or using regexp.

Comment (psfinaki) [reply docs/mutations.md:168]: Right. I think the latter-most form is clearer.

File: src/Stryker.Core/Stryker.Core/Mutants/CsharpMutantOrchestrator.cs (hunk 2 above)
Comment (rouke-broersma) [reply src/Stryker.Core/Stryker.Core/Mutants/CsharpMutantOrchestrator.cs:55]: https://github.com/stryker-mutator/stryker-net/blob/master/src/Stryker.Core/Stryker.Core/Mutators/MutatorBase.cs#L11

:) If it makes more sense, you can implement IMutator directly.

File: docs/mutations.md (hunk 1 above)
Comment (yakivyusin) [reply docs/mutations.md]: Rewritten using regexp

File: src/Stryker.Core/Stryker.Core/Mutants/CsharpMutantOrchestrator.cs (hunk 2 above)
Comment (yakivyusin) [reply src/Stryker.Core/Stryker.Core/Mutants/CsharpMutantOrchestrator.cs]: Refactored to `StringEmptyMutator`

File: docs/mutations.md (hunk 1 above)
Comment (rouke-broersma) [reply docs/mutations.md]: Mhm not sure I agree @psfinaki , `x.Trim() != ""` is probably way more clear than regex, to most people.

Comment (psfinaki) [reply docs/mutations.md]: Damn I actually meant `x.Trim() != ""`🤦Probably my eyes naturally filter out anything about regex, anyway sorry for the confusioin.

Comment (yakivyusin) [reply docs/mutations.md]: Refactored with `x.Trim()` )

**Pull request merged**